woodwork/schema.py

```diff
@@ -158,9 +268,12 @@ def _create_columns(self,
         """
         columns = {}
         for name in column_names:
-            semantic_tags_for_col = (semantic_tags or {}).get(name)
+            semantic_tags_for_col = _convert_input_to_set((semantic_tags or {}).get(name),
+                                                          error_language=f'semantic_tags for {name}')
+            _validate_not_setting_index_tags(semantic_tags_for_col, name)
```

Comment:
Removed `_validate_tags` from the schema_column.py file because it shouldn't have to care at all about index tags.